module/behaviour/strategy/PS3Walk/src/PS3Walk.cpp

```diff
@@ -17,123 +17,144 @@
-                                        Eigen::Vector3d(1, 0, 0),  // vector pointing forward relative to robot
-                                        LimbID::RIGHT_LEG)));
-                                }
-                                break;
-                        }
+                    else {
+                        log<NUClear::INFO>("Start walking");
+                        moving = true;
                     }
                 }
-                // If it's closed then we should try to reconnect
-                else if (!joystick.valid()) {
-                    joystick.reconnect();
+            });
+
+            on<Trigger<HeadControlEvent>>().then([this](const HeadControlEvent& event) {
+                if (event.pressed) {
+                    if (headLocked) {
+                        log<NUClear::INFO>("Head unlocked");
+                        headLocked = false;
+                    }
+                    else {
+                        log<NUClear::INFO>("Head locked");
+                        headLocked = true;
+                    }
+                }
+            });
+
+            on<Trigger<TriggerActionEvent>>().then([this](const TriggerActionEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Triggering actions");
+                    emit(std::make_unique<MotionCommand>(utility::behaviour::StandStill()));
+                    emit(std::make_unique<ActionPriorites>(ActionPriorites{id, {90}}));
+                    emit(std::make_unique<ExecuteScriptByName>(id, actions));
+                }
+            });
+
+            on<Trigger<ScriptTriggerEvent>>().then([this](const ScriptTriggerEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Standing");
+                    emit(std::make_unique<MotionCommand>(utility::behaviour::StandStill()));
+                    emit(std::make_unique<ActionPriorites>(ActionPriorites{id, {90}}));
+                    emit(std::make_unique<ExecuteScriptByName>(id, "Stand.yaml"));
+                }
+            });
+
+            on<Trigger<LeftKickEvent>>().then([this](const LeftKickEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Requesting Left Front Kick");
+                    // vector pointing forward relative to robot
+                    emit(std::make_unique<KickScriptCommand>(
+                        KickScriptCommand{Eigen::Vector3d(1.0, 0.0, 0.0), LimbID::LEFT_LEG}));
+                }
+            });
+
+            on<Trigger<RightKickEvent>>().then([this](const RightKickEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Requesting Right Front Kick");
+                    // vector pointing forward relative to robot
+                    emit(std::make_unique<KickScriptCommand>(
+                        KickScriptCommand(Eigen::Vector3d(1.0, 0.0, 0.0), LimbID::RIGHT_LEG)));
```

Comment:
Lol, this one is using a constructor and the other is using uniform initialisation (`{}` on 127 and `()` on 136)